cmd/private-prow-configs-mirror/main.go

```diff
@@ -0,0 +1,441 @@
+		}
+	}
+}
+
+func injectPrivateApprovePlugin(approves []plugins.Approve, orgRepos orgReposWithOfficialImages) {
+	logrus.Info("Processing...")
+
+	for index, approve := range approves {
+		repos := sets.NewString(approve.Repos...)
+
+		for _, orgRepo := range approve.Repos {
+			if orgRepos.isOfficialRepoFull(orgRepo) {
+				repo := strings.Split(orgRepo, "/")[1]
+
+				logrus.WithField("repo", repo).Info("Found")
+				repos.Insert(privateOrgRepo(repo))
+			}
+		}
+
+		approves[index].Repos = repos.List()
+	}
+}
+
+func injectPrivateLGTMPlugin(lgtms []plugins.Lgtm, orgRepos orgReposWithOfficialImages) {
+	logrus.Info("Processing...")
+
+	for index, lgtm := range lgtms {
+		repos := sets.NewString(lgtm.Repos...)
+
+		for _, orgRepo := range lgtm.Repos {
+			if orgRepos.isOfficialRepoFull(orgRepo) {
+				repo := strings.Split(orgRepo, "/")[1]
+
+				logrus.WithField("repo", repo).Info("Found")
+				repos.Insert(privateOrgRepo(repo))
+			}
+		}
+
+		lgtms[index].Repos = repos.List()
+	}
+}
+
+func injectPrivateBugzillaPlugin(bugzillaPlugins plugins.Bugzilla, orgRepos orgReposWithOfficialImages) {
+	logrus.Info("Processing...")
+
+	privateRepos := make(map[string]plugins.BugzillaRepoOptions)
+	for org, orgValue := range bugzillaPlugins.Orgs {
+		for repo, value := range orgValue.Repos {
+			if orgRepos.isOfficialRepo(org, repo) {
+				logrus.WithField("repo", repo).Info("Found")
+				privateRepos[repo] = value
+			}
+		}
+	}
+
+	if len(privateRepos) > 0 {
+		bugzillaPlugins.Orgs[openshiftPrivOrg] = plugins.BugzillaOrgOptions{Repos: privateRepos}
+	}
+}
+
+func injectPrivatePlugins(plugins map[string][]string, orgRepos orgReposWithOfficialImages) {
+	logrus.Info("Processing...")
+
+	var allPrivate []string
+
+	privateReposWithValues := make(map[string][]string)
+
+	for orgRepo, value := range plugins {
+		if orgRepos.isOfficialRepoFull(orgRepo) {
+			newValue := value
+
+			// We want to append the org level values, because the private repo
+			// can exist in different org with different configuration.
+			org := strings.Split(orgRepo, "/")[0]
+			if _, ok := plugins[org]; ok {
+				newValue = append(newValue, plugins[org]...)
+			}
+
+			privateReposWithValues[privateOrgRepo(strings.Split(orgRepo, "/")[1])] = newValue
+			allPrivate = append(allPrivate, newValue...)
+		}
+	}
+
+	mergeCommonPrivatePlugins(plugins, privateReposWithValues, getRepeatedValues(allPrivate))
+}
+
+// mergeCommonPrivatePlugins detects the common values from all the private repositories
+// and merges them into the org `openshift-priv` level.
+// In addition, it generates a repo. if it contains an extra non-common field.
+func mergeCommonPrivatePlugins(plugins map[string][]string, privateReposWithValues map[string][]string, repeatedValues sets.String) {
+	privateOrgPlugins := sets.NewString()
+
+	for repoName, repoValues := range privateReposWithValues {
+		repoLevelPlugins := sets.NewString()
+		valuesSet := sets.NewString(repoValues...)
+
+		if len(repeatedValues) > 0 {
```

Comment:
If there are none, why call this function at all?

Reply:
If there are none we need only to create the repo level, otherwise we will merge the common values under the org level (openshift-priv)